File: backend-alquiler-canchas/src/main/java/com/example/backend_alquiler_canchas/dto/ReservaDTO.java
 package com.example.backend_alquiler_canchas.dto;

import com.example.backend_alquiler_canchas.model.EstadoReserva;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ReservaDTO {

    private Integer idReserva;
    
    private LocalDate fechaReserva;
    
    private LocalTime horaInicio;
    
    private BigDecimal costoTotal;
    
    private BigDecimal adelanto;
    
    private EstadoReserva estado; 
    
    private Integer idCliente;   
    
    private Integer idCanchaDeporte;   
}

File: frontend-alquiler-canchas/src/app/models/reserva.dto.ts
export interface ReservaDTO {
  idReserva?: number;
  fechaReserva: string; // formato 'YYYY-MM-DD'
  horaInicio: string;   // formato 'HH:mm:ss'
  costoTotal: number;
  adelanto: number;
  idCliente: number;
  idCanchaDeporte: number;
  estado: 'PENDIENTE' | 'COMPLETADO' | 'INCOMPLETO';
}
